src/main/java/com/jcabi/ssh/AbstractSSHShell.java

```diff
@@ -0,0 +1,114 @@
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.ssh;
+
+import com.jcraft.jsch.Session;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.apache.commons.lang3.Validate;
+
+/**
+ * Common module for any ssh shell.
+ * @author Jose Volmei Dal Pra Junior (jrdalpra@gmail.com)
+ * @version $Id$
+ * @since 2.0
+ */
+@ToString
+@EqualsAndHashCode(of = { "addr", "port", "login" })
+@Getter
+@SuppressWarnings({ "PMD.UnusedPrivateField", "PMD.SingularField" })
+abstract class AbstractSSHShell implements Shell {
+
+    /**
+     * IP address of the server.
+     */
+    private final transient String addr;
+
+    /**
+     * Port to use.
+     */
+    private final transient int port;
+
+    /**
+     * User name.
+     */
+    private final transient String login;
+
+    /**
+     * Constructor.
+     * @param adr Address that you want to connect to.
+     * @param prt Port that you want to reach.
+     * @param user User that will be used when connecting.
+     * @throws UnknownHostException when host is unkwon.
+     */
+    public AbstractSSHShell(
+        final String adr,
+        final int prt,
+        final String user) throws UnknownHostException {
+        this.addr = InetAddress.getByName(adr).getHostAddress();
+        this.port = prt;
+        this.login = user;
+        Validate.matchesPattern(
+            this.addr,
+            "\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}",
+            "Invalid IP address of the server `%s`",
+            this.addr
+        );
+        Validate.notEmpty(user, "user name can't be empty");
+    }
+
+    // @checkstyle ParameterNumberCheck (6 lines)
```

Comment:
@jrdalpra you can use 
`@checkstyle ParameterNumberCheck (2 lines)` 
instead of 
`@checkstyle ParameterNumberCheck (6 lines)`